apps/liquidation-bot/src/lenders/FlashLender.ts

```diff
@@ -0,0 +1,61 @@
+import { BigNumber, providers } from 'ethers';
+import { FlashLiquidation, IFlashLoanProvider } from '../types';
+import { Network, getFlashLender } from '@notional-finance/util';
+import {
+  FlashLiquidator__factory,
+  FlashLenderWrapper__factory,
+} from '@notional-finance/contracts';
+
+const flashLiquidatorInterface = FlashLiquidator__factory.createInterface();
+
+export default class FlashLenderProvider implements IFlashLoanProvider {
+
+  constructor(
+    public network: Network,
+    public flashLiquidatorAddress: string,
+    public provider: providers.Provider
+  ) {
+  }
+
+  public async estimateGas(flashLiq: FlashLiquidation): Promise<BigNumber> {
+    const flashLender = FlashLenderWrapper__factory.connect(
+      getFlashLender({ network: this.network, token: flashLiq.flashBorrowAsset }),
+      this.provider
+    );
+
+    return flashLender.estimateGas.flash(
+      this.flashLiquidatorAddress,
+      flashLiq.flashBorrowAsset,
+      flashLiq.accountLiq.flashLoanAmount,
+      flashLiq.accountLiq.liquidation.encode(
+        flashLiq.accountLiq.accountId,
+        true,
+        flashLiq.preLiquidationTrade,
+        flashLiq.collateralTrade
```

Comment:
See the comment here:
https://github.com/notional-finance/contracts-v3/pull/42/files#r1619350689

I think given the changes we won't need `preLiquidationTrade` anymore either.